File: PilluleBoxApp/PilluleBox/app/src/main/java/Models/ScheduleModes/SingleMode.java
package Models.ScheduleModes;

import java.io.Serializable;
import java.util.Date;

public class SingleMode implements Serializable {
    private int id;
    private String medicineName;
    private Date dispensingDate;

    public SingleMode(String medicineName, Date dispensingDate) {
        this.medicineName = medicineName;
        this.dispensingDate = dispensingDate;
    }
    public SingleMode(int id, String medicineName, Date dispensingDate) {
        this.id = id;
        this.medicineName = medicineName;
        this.dispensingDate = dispensingDate;
    }

    // Getters y Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public String getMedicineName() { return medicineName; }
    public void setMedicineName(String medicineName) { this.medicineName = medicineName; }
    public Date getDispensingDate() { return dispensingDate; }
    public void setDispensingDate(Date dispensingDate) { this.dispensingDate = dispensingDate; }
}

File: PilluleBoxESP32/PilluleBox/SingleMode.cpp
#include "pillulebox.h"

SingleMode::SingleMode(int id) {
  this->id = id;
}

SingleMode::~SingleMode() {}

int SingleMode::getId() const {
  return id;
}

const char* SingleMode::getMedicineName() const {
  return medicine_name;
}

time_t SingleMode::getDispensingDate() const {
  return dispensing_date;
}

void SingleMode::setId(int id) {
  this->id = id;
}

void SingleMode::setMedicineName(const char* name) {
  strncpy(medicine_name, name, sizeof(medicine_name) - 1);
  medicine_name[sizeof(medicine_name) - 1] = '\0';
}

void SingleMode::setDispensingDate(time_t date) {
  dispensing_date = date;
}